transforms.py

```diff
@@ -81,6 +81,52 @@ def classification_simple_pipeline_builder(*, input_type, api_version):
     return Pipeline(pipeline)
 
 
+def classification_complex_pipeline_builder(*, input_type, api_version):
+    if input_type == "Datapoint" and api_version == "v1":
+        return None
+
+    if api_version == "v1":
+        transforms = transforms_v1
+        MaybeContiguous = ToContiguousV1
+        RandomResizedCropWithoutResize = RandomResizedCropWithoutResizeV1
+    elif api_version == "v2":
+        transforms = transforms_v2
+        MaybeContiguous = ToContiguousV2
+        RandomResizedCropWithoutResize = RandomResizedCropWithoutResizeV2
+    else:
+        raise RuntimeError(f"Got {api_version=}")
+
+    pipeline = []
+
+    if input_type in {"Tensor", "Datapoint"}:
+        pipeline.append(MaybeContiguous())
+
```

Comment:
For consistency with https://github.com/pytorch/vision/pull/7220/files and with the way the v2 transforms will be used (at least for now because the dataset wrapper returns PIL images), I believe we should convert to Tensor or to Datapoint here, within the transforms. Not prior to it.